### ISSUE TITLE
Unable to print: No permission to use port undefined. 

### ISSUE THREAD

**Issue by gkeraunen** [nobuhito/vscode.printcode#33]
I am running latest version 3.0.0 on Windows 7 with Chrome as default browser and always get the same error when trying to print:
> Unable to print: No permission to use port undefined. Please set different port number in User Settings: printcode.webServerPort and Reload Window.

I have tried different port numbers in the settings but nothing changes. Chrome always says: 
> Cannot print: localhost refused to connect.

I am not running any web server. Is this required in order to print?

**Comment by PeterWone** [comment]
The extension has an embedded webserver. It works by spawning a browser with the URL of its embedded webserver, which then delivers markdown rendered as HTML.

This problem occurs because for some reason the extension is getting a value of undefined when it reads your settings.

I don't think this repo is maintained. There are a number of unmerged PRs and the last commit was in October 2018. That's why I took the basic premise and wrote my own VS Code Printing extension. In the process I have addressed every outstanding issue and greatly improved the user experience.

**Comment by gkeraunen** [comment]
Is your extension available for download?
I just want an editor that can print. I like VS Code but have to copy and paste text into UltraEdit to print.

**Comment by PeterWone** [comment]
Yes. It would be extremely bad manners to advertise it here but if you [search for "printing" in the marketplace](https://marketplace.visualstudio.com/search?term=printing&target=VSCode&category=All%20categories&sortBy=Relevance) I'm sure you'll figure it out. It's _not_ listed in my name.
